app/controllers/api/v3/courses_controller.rb

```diff
@@ -24,11 +28,21 @@ def show
 
     private
 
+      def allowed_to_disable_pagination?
+        fields_for_sitemap?
+      end
+
+      def fields_for_sitemap?
+        if params[:fields] && (courses = params[:fields][:courses])
```

Comment:
this line has me well confused

Is it checking if the courses exists in the fields param and then also assigning it in the condition? if so:

`if params.dig(:fields, :courses)` would work? it'd return nil if it doesn't exist which would fail the check and then assign the variable inside the body of the if, otherwise this line seems heckin' complex

Reply:
agreed, i'll make those changes now